### DISCUSSION TITLE
PDF Reflow Will Not Be Added? Is it because mutools?

### DISCUSSION THREAD

**Discussion by benyaminl** [sumatrapdfreader/sumatrapdf#2945] [General]
Hello, I just want to ask a dumb question related to https://github.com/sumatrapdfreader/sumatrapdf/issues/1796

Does it really no plan for PdfReflow in the future? Because I only see sumatrapdf as the only PDF reader that's light on CPU, and power saving, than it's counterpart... 

Just asking, if it's not in plan, not possible in future, it's okay. Thanks

**Comment by GitHubRulesOK** [comment]
Not so dumb, as Acrobat do something along those lines in their XFA, however that is not and will not be part of standardized PDFs v1 or 2

You need to understand PDF is a fixed page format as paper cannot be stretched in a printer, sadly Adobe kept adding whistles and bells to a static page layout. PDF is a graph(ic) chart, the text is not considered as words and text it is single characters at X&Y but for convenience the plain text that they use for composition may be shown and extracted line by line.

A common issue is how could you move one bunch of characters from  start of one line up to end of another even if those are a b c and there's room to not exceed the void that looks like a marginal colorless un-inked area, then move the remaining blobs to the left and move perhaps almost as many up from line below. the problem the other way is letters fall of the turtles back at the bottom of the world. Just like no such thing as line wrap nor is it possible to carry over a line from page to page they are discreetly independent unlike the html you might print a pdf from.

So word editors do that very easily maintaining multiple spaces between lines etc. but try doing that in a CAD or graphics package and its more difficult. Yes its done well by heavier pdf apps even lightweight Tracker xchange, but it needs dedicated editing code.
